core/services/functions/external_adapter_client.go

```diff
@@ -190,7 +197,11 @@ func (ea *externalAdapterClient) request(
 	req.Header.Set("Content-Type", "application/json")
 
 	start := time.Now()
-	client := &http.Client{}
+	retryClient := retryablehttp.NewClient()
```

Comment:
Maybe add a comment emphasizing that re-tries happen only on connection errors and 5XX codes (except 501).